app/questionnaire/questionnaire_schema.py

```diff
@@ -974,22 +976,26 @@ def _get_section_and_block_ids_dependencies_for_progress_source_and_answer_ids_f
                 if identifier != current_section_id:
                     dependencies_ids_for_progress_value_source["sections"][
                         identifier
-                    ] = [current_section_id]
+                    ] = OrderedSet()
+                    # Type ignore: Added as this will be a set rather than a dict at this point
+                    dependencies_ids_for_progress_value_source["sections"][identifier].add(current_section_id)  # type: ignore
             elif selector == "block" and (
                 section_id := self.get_section_id_for_block_id(identifier)
             ):
                 # Type ignore: The identifier key will return a list
                 if section_id != current_section_id:
                     dependencies_ids_for_progress_value_source["blocks"][section_id] = {
-                        identifier: []
+                        identifier: OrderedSet()
                     }
-                    dependencies_ids_for_progress_value_source["blocks"][section_id][identifier].append(current_section_id)  # type: ignore
+                    dependencies_ids_for_progress_value_source["blocks"][section_id][identifier].add(current_section_id)  # type: ignore
 
         return answer_id_list, dependencies_ids_for_progress_value_source
 
     def _get_rules_section_dependencies(
         self, current_section_id: str, rules: Mapping | Sequence
-    ) -> tuple[set[str], dict[str, list[str]], dict[str, dict[str, list[str]]]]:
+    ) -> tuple[
+        set[str], dict[str, OrderedSet[str]], dict[str, dict[str, OrderedSet[str]]]
```

Comment:
given how many occurrences of `dict[str, OrderedSet[str]]` there are, it seems like a good candidate for a TypeAlias, just thinking it might make the type hints shorter and more readable

Reply:
Updated 👍